### DISCUSSION TITLE
Pug enhancement

### DISCUSSION THREAD

**Discussion by louiss0** [SubZtep/vite-plugin-pug#8] [Ideas]
I'm grateful for the fact that you created this for us but this plugin doesn't make it worthwhile to use layouts at all. The point of pug is to abstract away writing html boilerplate code. I could use the extends feature but i now have to create the layout for pug and the html page. I'm basically stuck using pug files as components.  I need a way to be able to write pug inside the tags of the pug tag and say that this file extends the template.     
![2022-04-13](https://user-images.githubusercontent.com/29028601/163171955-05029cb0-c0ce-4bf4-8e3c-bc78533f1
![2022-04-13 (1)](https://user-images.githubusercontent.com/29028601/163171974-1b49d20d-6b86-441f-b082-83e80d8f31fa.png)
7b5.png)

**Comment by SubZtep** [comment]
Yes, very minimal. I was using it with [A-Frame](https://aframe.io/) where I built a component hierarchy with Pug's includes and mixins, it was useful in that case.

I want to upgrade at some point and this enhancement should be a part of it. I made a [Babel macro for inline pug](https://github.com/SubZtep/pug.macro) embed and I use it in web components, works pretty much the same.
